### PR TITLE
Feature/ssl conf

### LINKED ISSUE #205: Apache template has wrong paths for SSL Cert Files
The Apache webserver config templates hardcode the location of the SSL certificate files:

```
  SSLCertificateFile /etc/apache2/ssl/<%= @application[:domains].first %>.crt
  SSLCertificateKeyFile /etc/apache2/ssl/<%= @application[:domains].first %>.key
  <% if @application[:ssl_configuration][:chain] -%>
  SSLCACertificateFile /etc/apache2/ssl/<%= @application[:domains].first %>.ca
```

However, `Drivers::Webservers::Apache2#conf_dir` defines the write dynamically depending on the platform family:

```
def conf_dir
  File.join('/', 'etc', node['platform_family'] == 'debian' ? 'apache2' : 'httpd')
end
```

So on an AMI the end result is that the deploy fails because the config file expects the certificate in `/etc/apache2/ssl/...`, but the Apache2 driver is writing the certificate files to `/etc/httpd/ssl/...`

So I'll submit a PR for this accordingly. I'm thinking I should add the end location to the output hash in the driver so we don't need to repeat that information repeatedly:

```
SSLCertificateFile <%= output[:ssl_dir] %>/<%= @application[:domains].first %>.crt
```

instead of doing this ugly hack:

```
SSLCertificateFile /etc/<%= node['platform_family'] == 'debian' ? 'apache2' : 'httpd' %>/ssl/<%= @application[:domains].first %>.crt
```

### PR DESCRIPTION
**Pull request by richseviora**
This PR makes the following changes, fixes #205.

- Adds `ssl_cert_dir` to the Apache 2 webserver output.
- Updates the Passenger and Upstream versions of the Apache 2 template to use `ssl_cert_dir` for the SSL certificate paths.
- Adds the `X-Forwarded-Proto https` header to the Passenger template.

### CODE REVIEW

File: .ruby-gemset [hunk 1]
```diff
@@ -1 +1 @@
-opsworks_ruby
+-global
```
Comment (ajgon) [inline .ruby-gemset]: Why this change?

File: libraries/drivers_webserver_apache2.rb [hunk 2]
```diff
@@ -26,7 +26,8 @@ def self.passenger_supported?
 
       def settings
         output = super.merge(
-          log_dir: node['deploy'][app['shortname']][driver_type]['log_dir'] || "/var/log/#{service_name}"
+          log_dir: node['deploy'][app['shortname']][driver_type]['log_dir'] || "/var/log/#{service_name}",
+          ssl_cert_dir: ssl_cert_dir
```
Comment (ajgon) [inline libraries/drivers_webserver_apache2.rb]: There are two conflicting ideas here. `output` is supposed to contain only options configurable by user. With this approach, you "hardcode" `ssl_cert_dir` despite allowing it to be configured in line 11 of this file. So I suggest one of two options:

* Make directory configurable
  * It basically means altering this line to: `ssl_cert_dir: node['deploy'][app['shortname']][driver_type]['ssl_cert_dir'] || default_ssl_cert_dir` and renaming the method to `default_ssl_cert_dir` to avoid confusion.
  * Please, don't forget to add this option to documentation if you choose this path.
* Keep the original concept:
  * remove `ssl_cert_dir` from filter options in line 11
  * remove this line
  * in file `libraries/drivers_webserver_base.rb` in `add_appserver_config` method add another key-value pair to `opts` variable:
    ```ruby
    opts = { application: app, deploy_dir: deploy_dir(app), out: out, conf_dir: conf_dir, adapter: adapter, name: a.adapter, deploy_env: deploy_env, appserver_config: a.webserver_config_params, ssl_cert_dir: ssl_cert_dir }
    ```
  * and last, but not least - alter the templates with `@ssl_cert_dir` instead of `out[:ssl_cert_dir)`

File: .ruby-gemset (hunk 1 above)
Comment (richseviora) [reply .ruby-gemset]: Oops! Fixed!

File: libraries/drivers_webserver_apache2.rb (hunk 2 above)
Comment (richseviora) [reply libraries/drivers_webserver_apache2.rb]: Second option sounds easier; I can't think of any immediate reason to make the SSL dir configurable.

Comment (richseviora) [reply libraries/drivers_webserver_apache2.rb]: So that's been done + I'd like to capture the architectural direction about `output` somewhere. Any recommendations? I was thinking perhaps `Drivers::DSL::Output`?

**Comment by richseviora** [comment]
I've addressed the PR feedback as follows:

- Moved `ssl_cert_dir` to the app server template config options and adjusted the Apache templates accordingly.
- Applied the same change to the Nginx template.
- Squashed commits & reworded, though I'm going to add more content in the commit body now.

File: .gitignore [hunk 3]
```diff
@@ -23,3 +23,8 @@ docs/build/*
 vendor
 client.pem
 .chef.login
+
+# Ignore Jetbrains config
+.idea
+# Ignore VS Code
+.vscode
```
Comment (ajgon) [inline .gitignore]: Please, don't pollute project `.gitignore` with this files, and use [global](https://help.github.com/en/articles/ignoring-files#create-a-global-gitignore) or [project local](https://help.github.com/en/articles/ignoring-files#explicit-repository-excludes) ignore file instead.

File: .ruby-gemset [hunk 4]
```diff
@@ -1 +1 @@
-opsworks_ruby
+-opsworks_ruby
```
Comment (ajgon) [inline .ruby-gemset]: I assume this `-` (minus) sign got there by accident? 😄

**Comment by richseviora** [comment]
Updated :)
